### DISCUSSION TITLE
GitHub Copilot 无法正常使用

### DISCUSSION THREAD

**Discussion by lijiongquan666** [community/community#17550] [Copilot Conversations] [answered]
无法连接服务器   有没有大佬知道怎么弄？ 改了DNS 改了hosts文件  还是不行                   会出现以下提示  ：
GitHub Copilot could not connect to server. Extension activation failed: "connect ECONNREFUSED 127.0.0.1:443"    查了一天没查到解决方法  求大佬解答····

**Comment by katorly** [comment, marked as answer]
It is the magical network & network operators are to blame.  

Try modifying your computer's DNS address 2 solve this problem:  
- Change ipv4 into `8.8.8.8`
- Change ipv6 into `2001:4860:4860::8888`

I have used this method, and it really helps! If this helps, please click `Mark as answer`.  

---

是国内的魔幻网络和运营商搞的鬼。  

尝试修改您电脑的DNS服务器地址以解决此问题：
- 将ipv4改为`8.8.8.8`
- 将ipv6改为`2001:4860:4860::8888`

我使用了此方法，并且问题得到了解决。如果答案对您有帮助，请点击`Mark as answer`，以表示此问题得到了解决。

**Reply by katorly** [reply]
What you've said in the post, "I have modified DNS": You may forget to modify your ipv6 DNS address.  

If these doesn't help, please turn to the command line of your computer and type `nslookup github.com`, and paste the response here. I'll try my best 2 help!

---

对于您在帖子中提到的“改了DNS”：可能是由于您忘记改ipv6地址了。  

如果这些都未能解决您的问题，请转到您电脑的命令行，并输入`nslookup github.com`，然后把得到的文字贴出来。我会尽全力帮忙（解决此问题）！

**Reply by lijiongquan666** [reply]
非常感谢   用了你的方法后成功了     我就是忘记修改ipv6 DNS地址了   改了ipv6 DNS地址后成功可以使用了 非常感谢！！！

**Reply by limingzhou23** [reply]
您好 请问一个问题
➜  ~ nslookup github.com
Server:		8.8.8.8
Address:	8.8.8.8#53

Non-authoritative answer:
Name:	github.com
Address: 20.205.243.166

我修改了DNS地址 但是仍然报错“connect ECONNREFUSED 127.0.0.1:443” 这是我的命令行输入nslookup github.com的结果 请问应该如何解决此问题呢？

**Reply by katorly** [reply]
> 您好 请问一个问题 ➜ ~ nslookup github.com Server: 8.8.8.8 Address: 8.8.8.8#53
> 
> Non-authoritative answer: Name: github.com Address: 20.205.243.166
> 
> 我修改了DNS地址 但是仍然报错“connect ECONNREFUSED 127.0.0.1:443” 这是我的命令行输入nslookup github.com的结果 请问应该如何解决此问题呢？

你不用ipv6吗？我之前在电脑上用ipv6的服务器可以

**Reply by dengchunyu** [reply]
nslookup github.com
服务器:  UnKnown
Address:  2001:4860:4860::8888

*** UnKnown 找不到 github.com: No response from server
我这里改了DNS，仍然不行，唉

**Reply by ceasarXuu** [reply]
I've set up DNS addresses for IPv4 and IPv6 separately but it still doesn't work.

```
[INFO] [default] [2023-10-18T08:46:48.779Z] [fetchCompletions] engine http://localhost:5001/v1/engines/copilot-codex
[ERROR] [ghostText] [2023-10-18T08:46:48.786Z] Error on ghost text request: (FetchError) connect ECONNREFUSED 127.0.0.1:5001

Server:         8.8.8.8
Address:        8.8.8.8#53

Non-authoritative answer:
Name:   github.com
Address: 20.205.243.166
```

I use the company's network and cannot turn on IPv6

**Reply by katorly** [reply]
> I've set up DNS addresses for IPv4 and IPv6 separately but it still doesn't work.
> 
> ```
> [INFO] [default] [2023-10-18T08:46:48.779Z] [fetchCompletions] engine http://localhost:5001/v1/engines/copilot-codex
> [ERROR] [ghostText] [2023-10-18T08:46:48.786Z] Error on ghost text request: (FetchError) connect ECONNREFUSED 127.0.0.1:5001
> 
> Server:         8.8.8.8
> Address:        8.8.8.8#53
> 
> Non-authoritative answer:
> Name:   github.com
> Address: 20.205.243.166
> ```
> 
> I use the company's network and cannot turn on IPv6 

这个方法可能已经用不了了，最近国内的网络环境是怎么样的相信大家都知道

**Reply by ceasarXuu** [reply]
我用自己的代理也不行，试了好几个节点了
```
[DEBUG] [proxy-socket-factory] [2023-10-18T09:19:00.720Z] Attempting to establish connection to proxy
[DEBUG] [proxy-socket-factory] [2023-10-18T09:19:00.724Z] Socket Connect returned status code,405
[ERROR] [ghostText] [2023-10-18T09:19:00.726Z] Error on ghost text request: (FetchError) tunneling socket could not be established, statusCode=405
```

**Reply by katorly** [reply]
> 我用自己的代理也不行，试了好几个节点了
> ```
> [DEBUG] [proxy-socket-factory] [2023-10-18T09:19:00.720Z] Attempting to establish connection to proxy
> [DEBUG] [proxy-socket-factory] [2023-10-18T09:19:00.724Z] Socket Connect returned status code,405
> [ERROR] [ghostText] [2023-10-18T09:19:00.726Z] Error on ghost text request: (FetchError) tunneling socket could not be established, statusCode=405
> ```

我这里梯子一直可以，不清楚您那边的情况哦

**Comment by liaoweijie12** [comment]
> 

这个可以https://www.le9plus7.top/articles/61
